src/main/webapp/app/exercises/shared/statistics/doughnut-chart.component.ts

```diff
@@ -101,4 +85,27 @@ export class DoughnutChartComponent implements OnChanges, OnInit {
             this.router.navigate(this.titleLink);
         }
     }
+
+    /**
+     * Assigns a given array of numbers to ngxData
+     * @param values the values that should be displayed by the chart
+     * @private
+     */
+    private assignValuesToData(values: number[]) {
+        this.ngxDoughnutData[0].value = values[0];
+        this.ngxDoughnutData[1].value = values[1];
+        this.ngxDoughnutData.forEach((entry: NgxDataEntry, index: number) => (entry.value = values[index]));
+        this.ngxDoughnutData = [...this.ngxDoughnutData];
+    }
+
+    /**
+     * Modifies the tooltip content of the chart.
+     * @param value the default tooltip content that has to be replaced
+     * @returns absolute value represented by doughnut piece or 0 if the currentMax is 0.
+     * This is necessary in order to compensate the workaround for
+     * displaying a chart even if no values are there to display (i.e. currentMax is 0)
+     */
+    valueFormatting(value: any): string {
```

Comment:
same as above, `value: any` is too vague IMO. If the `value` is a tooltip content, isn't it of type `string`? And if `value` only wraps the actual content in `value.value` but we do not need to access any other property of `value`, why don't you change the method to:
````
formatChartTooltips(defaultTooltip: string): string {
    return this.currentMax === 0 ? '0' : defaultTooltip;
}
````
The method call would be:
`bindFormatting = this.formatChartTooltips.bind(this.value);`

I am not sure if the passing `this.value` works but I think the naming at least is not specific enough for what the method actually does, i.e. modifying the tooltip.

Reply:
This method is passed to the framework where it is executed. Therefore, the arguments are filled by the framework. Unfortunately, the framework seems to pass a dedicated object to the method instead of just a string. In the screenshot below I modified the method so that it returns the string representation of `value`. 
![image](https://user-images.githubusercontent.com/80622272/144426856-edf9eeca-6fbf-4386-a687-9abc4f76904d.png)

I'm not quite sure if I understand the second half of your comment correctly. As far as I know `bind(this)` allows the framework to access attributes of the component the method is written in (which is necessary as we need `currentMax` in order to determine the tooltip content). Passing `this.value` would not work, as the component does not have an attribute named `value`.